### ISSUE TITLE
[RFC]: remove conflict with symfony/console

### ISSUE THREAD

**Issue by fengelniederhammer** [laminas/laminas-cache#188]
### Goal

Make laminas-cache compatible with symfony/console < 5.1

### Background

In the project, we use laminas-cache a lot, but we are still stuck to version 2. Also lots of our dev-dependencies come with symfony/console.

To prepare an upgrade to laminas-cache v3, we would like to upgrade to ^2.12.0 in all our artifacts, however this is blocked by the conflict with symfony/console < 5.1.

Is it possible to remove the conflict in https://github.com/laminas/laminas-cache/blob/2.12.x/composer.json#L64-L66?

This really troubles us a lot and keeps us stuck with even more outdated dependencies than we have already anyway.

**Comment by Ocramius** [comment]
See https://github.com/laminas/laminas-cache/commit/57b794c7ce1a50ec8f16c68ccceda1e498a5bc9e

If you are able to test the component with older `symfony/console` releases, we may be able to drop the conflict, which is added to prevent runtime incompatibilities.

These conflicts are generally added after careful evaluation, usually because supporting ranges wider than that would lead to crashes and errors.

**Comment by boesing** [comment]
I've added the conflict as `symfony/console` v5.1 introduced the `Command::FAILURE` and `Command::SUCCESS` constants which I do use in the command.

I think we are able to widen the version constraint by a bit. The main problem is, that at some point, we are unable to verify if the command is working or not. So we definitely have to add a conflict to that version we are actually testing on. And I am pretty sure that the version does not include v2.0 as that version is 10 years old and even our tooling (psalm for example) does not support that range of `symfony/console`. So what I can offer is to widen the constraint to `<4.2` but that won't fit in your requirements.

**Comment by Ocramius** [comment]
Given that `2.x` is supposed to be security-maintenance-only, and stability is king on that, and considering @boesing's last comment, I think this is to be kept as-is.

My suggestion to @fengelniederhammer is to keep a local company-specific fork of `laminas/laminas-cache:2.12.x`, with relaxed dependency constraints, and to operate with the expanded dependency range at their own risk, via `--ignore-platform-reqs` or whichever means possible, as long as the responsibility is local to their environment.

Closing here as `won't fix` - the way forward is to incrementally upgrade single components until you get there, and you will get there with a lot of work (like everyone else, heh).

**Comment by fengelniederhammer** [comment]
So from @boesing's comment I conclude, that you have an IMPLICIT dependency on symfony/console and only express that through a "conflict"? That seems questionable to me.

Using it is a valid argument then, but why not introduce it as a proper dependency? This would have prevented this ticket in the first place, because this makes it obvious to users, that it is an actualy requirement.

**Comment by boesing** [comment]
@fengelniederhammer because the command can only be used with laminas-cli but laminas-cli is not a hard requirement here. 

But you are right, it should be either explicit or a dedicated package. Will think about it and maybe extract this to a new package. However, this will only affect v3 of laminas-cache.
